sdk/dotnet/Pulumi/Config_Options.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright 2016-2018, Pulumi Corporation
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Pulumi
+{
+    public partial class Config
+    {
+        /// <summary>
+        /// StringConfigOptions may be used to constrain the set of legal values a string config value may contain.
+        /// </summary>
+        public class StringOptions
+        {
+            /// <summary>
+            /// The legal enum values. If it does not match, a ConfigEnumError is thrown.
+            /// </summary>
+            public ISet<string> AllowedValues = new HashSet<string>();
+
+            /// <summary>
+            /// The minimum string length. If the string is not this long, a ConfigRangeError is thrown.
+            /// </summary>
+            public int? MinLength;

```

Comment:
Should these be properties rather than fields?

Reply:
honestly, i don't care :)